portability-gateway/src/main/java/org/dataportabilityproject/gateway/reference/ReferenceApiModule.java

```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright 2018 The Data-Portability Project Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.dataportabilityproject.gateway.reference;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.multibindings.MapBinder;
+import com.sun.net.httpserver.HttpHandler;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import javax.inject.Named;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Bindings the reference api server.
+ */
+public class ReferenceApiModule extends AbstractModule {
+
+  private static final Logger logger = LoggerFactory.getLogger(ReferenceApiModule.class);
+
+  @Provides
+  @Named("httpPort")
+  public int httpPort() {
+    return 8080; // TODO: set with a flag
+  }
+
+  @Provides
+  @Named("defaultView")
+  public String defaultView() {
+    return "static/index.html"; // TODO: set with a flag
+  }
+
+  @Provides
+  @Named("httpExecutor")
+  public Executor executor(UncaughtExceptionHandler uncaughtExceptionHandler) {
+    ThreadFactory threadPoolFactory = new ThreadFactoryBuilder()
+        .setNameFormat("http-server-%d")
+        .setUncaughtExceptionHandler(uncaughtExceptionHandler)
+        .build();
+    return Executors.newCachedThreadPool(threadPoolFactory);
+  }
+
+  @Provides
+  public UncaughtExceptionHandler uncaughtExceptionHandler() {
+    return new UncaughtExceptionHandler() {
+      @Override
+      public void uncaughtException(Thread thread, Throwable t) {
+        logger.warn("Uncaught exception in thread: {}", thread.getName(), t);
+      }
+    };
+  }
+
+  @Override
+  protected void configure() {
+    MapBinder<String, HttpHandler> mapbinder
+        = MapBinder.newMapBinder(binder(), String.class, HttpHandler.class);
+
+    // HttpServer does exact longest matching prefix for context matching. This means
+    // /_/listServices, /_/listServicesthisshouldnotwork and /_/listServices/path/to/resource will
+    // all be handled by the ListServicesHandler below. To prevent this, each handler below should
+    // validate the request URI that it is getting passed in.
+    mapbinder.addBinding(DataTransferHandler.PATH).to(DataTransferHandler.class);
+
```

Comment:
Add TODO to hook up the below as they are implemented

Reply:
Done